### PR TITLE
dev-vcs/pijul: new package, add 1.0.0_beta4

### PR DESCRIPTION
**Pull request by laumann**

### CODE REVIEW

File: dev-vcs/pijul/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>t@laumann.xyz</email>
+		<name>Thomas Bracht Laumann Jespersen</name>
+	</maintainer>
```
Comment (laumann) [inline dev-vcs/pijul/metadata.xml:7]: I considered adding an `<upstream>` but decided against it for the moment.

File: dev-vcs/pijul/pijul-1.0.0_beta4.ebuild [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+BDEPEND="virtual/rust"
+
+src_prepare() {
+	default
+
+	# We get the sources from one of the crates, as the Pijul source is managed
+	# with Pijul on nest.pijul.com - we obviously can't clone it (because we'd
+	# need a working pijul binary to do that), and the Nest doesn't support
+	# fetching archives over HTTP or anything like that.
+	cp -af "${WORKDIR}/cargo_home/gentoo/${PN}-${MY_PIJUL_PV}" "${WORKDIR}" || die
```
Comment (laumann) [inline dev-vcs/pijul/pijul-1.0.0_beta4.ebuild:378]: I guess I could also set `S` appropriately and not override src_prepare() at all...

**Review by antecrescent — COMMENTED** [review]
You may want to follow the [information for contributors ](https://wiki.gentoo.org/wiki/Project:GURU/Information_for_Contributors) and submit your ebuild to the dev branch directly in order to get your changes promptly merged. :)

Apart from the dependency issue, this looks good to me!

File: dev-vcs/pijul/pijul-1.0.0_beta4.ebuild [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+	xattr-0.2.3
+	yasna-0.4.0
+	zbus-1.9.3
+	zbus_macros-1.9.3
+	zeroize-1.3.0
+	zeroize_derive-1.3.3
+	zstd-seekable-0.1.14
+	zvariant-2.10.0
+	zvariant_derive-2.10.0
+	${PN}-${MY_PIJUL_PV}
```
[the remark this answers is not in the document]
Comment (laumann) [reply dev-vcs/pijul/pijul-1.0.0_beta4.ebuild:347]: There's also the `.` between "beta" and "4" that needs to be considered - but I'll play around and see if I can't get rid of the variable

File: dev-vcs/pijul/pijul-1.0.0_beta4.ebuild [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+# Dependent crate licenses
+LICENSE+="
+	Apache-2.0 BSD-2 BSD CC0-1.0 GPL-2 GPL-2+ MIT Unicode-DFS-2016 ZLIB
+"
+SLOT="0"
+KEYWORDS="~amd64"
+S="${WORKDIR}/${PN}-${MY_PIJUL_PV}"
+
+IUSE="git"
+BDEPEND="virtual/rust"
```
[the remark this answers is not in the document]
Comment (laumann) [reply dev-vcs/pijul/pijul-1.0.0_beta4.ebuild:369]: Right, somehow I'd convinced myself that I still needed to add it in (just to make it explicit).

File: dev-vcs/pijul/pijul-1.0.0_beta4.ebuild [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+"
+
+inherit cargo
+
+DESCRIPTION="A distributed version control system."
+HOMEPAGE="https://pijul.org/ https://nest.pijul.com/pijul/pijul"
+SRC_URI="
+	$(cargo_crate_uris)
+"
+RESTRICT="mirror"
```
[the remark this answers is not in the document]
Comment (laumann) [reply dev-vcs/pijul/pijul-1.0.0_beta4.ebuild:357]: Good question - I'll remove it, thanks!

File: dev-vcs/pijul/pijul-1.0.0_beta4.ebuild [hunk 6]
```diff
@@ -0,0 +1,386 @@
...
+LICENSE="GPL-2+"
+# Dependent crate licenses
+LICENSE+="
+	Apache-2.0 BSD-2 BSD CC0-1.0 GPL-2 GPL-2+ MIT Unicode-DFS-2016 ZLIB
+"
+SLOT="0"
+KEYWORDS="~amd64"
+S="${WORKDIR}/${PN}-${MY_PIJUL_PV}"
+
+IUSE="git"
```
[the remark this answers is not in the document]
Comment (laumann) [reply dev-vcs/pijul/pijul-1.0.0_beta4.ebuild:368]: I guess it should be
```
DEPEND="git ( dev-libs/libgit2:= )"
```
but, yes, I'll fix it.

**Comment by laumann** [comment]
@antecrescent thanks for taking a look! I'll address your feedback shortly.

File: dev-vcs/pijul/pijul-1.0.0_beta4.ebuild (hunk 2 above)
Comment (laumann) [reply dev-vcs/pijul/pijul-1.0.0_beta4.ebuild:378]: done.

File: dev-vcs/pijul/pijul-1.0.0_beta4.ebuild (hunk 4 above)
[the remark this answers is not in the document]
Comment (laumann) [reply dev-vcs/pijul/pijul-1.0.0_beta4.ebuild:369]: fixed

File: dev-vcs/pijul/pijul-1.0.0_beta4.ebuild (hunk 6 above)
[the remark this answers is not in the document]
Comment (laumann) [reply dev-vcs/pijul/pijul-1.0.0_beta4.ebuild:368]: fixed

File: dev-vcs/pijul/pijul-1.0.0_beta4.ebuild (hunk 3 above)
[the remark this answers is not in the document]
Comment (laumann) [reply dev-vcs/pijul/pijul-1.0.0_beta4.ebuild]: I ended up leaving the variable in - I felt like it's too much mangling to get from "1.0.0_beta4" to "1.0.0-beta.4" (the dot between "beta" and "4" is unlucky).

File: dev-vcs/pijul/pijul-1.0.0_beta4.ebuild (hunk 6 above)
[the remark this answers is not in the document]
Comment (laumann) [reply dev-vcs/pijul/pijul-1.0.0_beta4.ebuild:365]: Thanks to @ionenwks awesome [qa-vdb](https://github.com/ionenwks/iwdevtools#qa-vdb), I've also added dev-libs/openssl as a rdep.

**Comment by stkw0** [comment]
Seems that beta6 is out, could you update it?

**Comment by laumann** [comment]
I could, kinda gave up on it :-) I have it here though: https://git.sr.ht/~laumann/portage/tree/master/item/dev-vcs/pijul/pijul-1.0.0_beta6.ebuild

**Pull request closed without merging**